ocs_ci/deployment/mce.py

```diff
@@ -0,0 +1,243 @@
+
+    def create_mce_subscription(self):
+        """
+        Creates subscription for mce operator
+
+        """
+        mce_subscription_yaml_data = templating.load_yaml(
+            constants.MCE_SUBSCRIPTION_YAML
+        )
+
+        if config.DEPLOYMENT.get("mce_latest_stable"):
+            mce_subscription_yaml_data["spec"][
+                "source"
+            ] = constants.OPERATOR_CATALOG_SOURCE_NAME
+            mce_sub_channel = "stable-2.7"
+
+        mce_subscription_yaml_data["spec"]["channel"] = f"{mce_sub_channel}"
+        mce_subscription_manifest = tempfile.NamedTemporaryFile(
+            mode="w+", prefix="mce_subscription_manifest", delete=False
+        )
+        templating.dump_data_to_temp_yaml(
+            mce_subscription_yaml_data, mce_subscription_manifest.name
+        )
+        logger.info("Creating subscription for mce operator")
+        run_cmd(f"oc create -f {mce_subscription_manifest.name}")
+        OCP(
+            kind=constants.SUBSCRIPTION_COREOS,
+            namespace=self.namespace,
+            resource_name=constants.MCE_OPERATOR,
+        ).check_resource_existence(
+            should_exist=True, resource_name=constants.MCE_OPERATOR
+        )
+
+    def check_hypershift_namespace(self):
+        """
+        Check hypershift namespace created
+
+        """
+        logger.info(f"hypershift namespace {self.namespace} was created successfully")
+        is_hypershift_ns_available = self.ns_obj.is_exist(
+            resource_name=constants.HYPERSHIFT_NAMESPACE,
+        )
+        return is_hypershift_ns_available
+
+    def check_supported_versions(self):
+        """
+        Check supported ocp versions for hcp cluster creation
+
+        """
+        configmaps_obj = OCP(
+            kind=constants.CONFIGMAP,
+            namespace=constants.HYPERSHIFT_NAMESPACE,
+        )
+
+        assert configmaps_obj.is_exist(
+            resource_name=constants.SUPPORTED_VERSIONS_CONFIGMAP
+        ), f"Configmap {constants.SUPPORTED_VERSIONS_CONFIGMAP} does not exist in hypershift namespace"
+
+        cmd = "oc get cm -n hypershift supported-versions -o jsonpath='{.data.supported-versions}'"
+        cmd_res = exec_cmd(cmd, shell=True)
+        if cmd_res.returncode == 0:
+            supported_versions = cmd_res.stdout.decode("utf-8")
+            logger.info(f"Supported versions: {supported_versions}")
+
+        if not get_running_ocp_version() in supported_versions:
+            self.create_image_override()
+
+    def create_image_override(self):
+        """
+        Create hypershift image override cm
+        """
+        # Create image override configmap using the image override json
+        cmd = (
+            f"oc create cm {self.hypershift_override_image_cm} --from-file={constants.IMAGE_OVERRIDE_JSON}"
+            "-n {self.namespace}"
+        )
+        cmd_res = exec_cmd(cmd, shell=True)
+        assert cmd_res.returncode == 0
```

Comment:
overall it its better not use assert in deployments, but in tests. This way we'll have errors in log showing that deployment failed, instead of assertion signaling about a product bug